### ISSUE TITLE
Aria-label strings are not rendered correctly

### ISSUE THREAD

**Issue by bmamey** [highcharts/highcharts#18931]
#### Expected behaviour
HTML safe strings should be rendered correctly in aria-label

#### Actual behaviour
Some punctuations in strings returned from `accessibility.point.descriptionFormatter` callback are being replaced with HTML entities. For example: `Foobar's test` is being rendered as `Foobar&#x27;s test` in aria-labels and this throws screen readers off. 

![Screenshot 2023-05-03 at 5 20 56 PM](https://user-images.githubusercontent.com/6766540/236061887-6432b267-2d3e-45f5-b9b0-a5d45de70d2e.png)

         

#### Live demo with steps to reproduce
[template:https://jsfiddle.net/2rfb1836/1/](https://jsfiddle.net/2rfb1836/1/)

#### Product version
Highcharts version `8.2.2`

#### Affected browser(s)
Chrome

I know upgrading to the latest version solves the issue, but I'd like to know if you all can get a patch in for v 8.x
